src/ipc/ipc4/handler.c

```diff
@@ -460,6 +528,24 @@ static int ipc4_set_pipeline_state(struct ipc4_message_request *ipc4)
 		}
 	}
 
+	for (i = 0; i < ppl_count; i++) {
+		ppl_icd = ipc_get_comp_by_ppl_id(ipc, COMP_TYPE_PIPELINE, ppl_id[i]);
+		if (!ppl_icd) {
+			ipc_cmd_err(&ipc_tr, "ipc: comp %d not found", ppl_id[i]);
+			return IPC4_INVALID_RESOURCE_ID;
+		}
+
+		/* Do state change only on current core.
+		 * Otherwise the IPC will be passed to target core in trigger
+		 * phase
```

Comment:
isn't that too late? What if the first pipeline runs on core 0, so it is configured here. The next pipeline runs on core 1, so it isn't configured yet. Then we continue below to trigger pipeline 0, while pipeline 1 is still unconfigured?

Reply:
right, we don't have a means to separately configure and trigger a pipeline on remote core, afaik.
If we would do the IDC_MSG_PPL_STATE here in the config phase, it will trigger non core 0 pipelines right away.
If we add another loop in between config and trigger loop then it is almost good except if we have two non core 0 pipelines directly connected
If we do it after the trigger loop, then we have similar issue.

I'm not sure how that could be solved other than modifying the IDC_MSG_PPL_STATE message and add support for decoupling of config and trigger phases.